### DISCUSSION TITLE
How to add a new custom env to run tests on that particular env.

### DISCUSSION THREAD

**Discussion by AvinashK133** [seleniumbase/SeleniumBase#2254] [Q&A] [answered]
Hi,
How can i add a new env other than the default ones, i tried adding in the constants.py file in the env class, but it gives below error upon execution.
I am adding new env name as "replica_qa"
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: argument --env: invalid choice: 'replica_qa' (choose from 'qa', 'rc', 'staging', 'develop', 'production', 'performance', 'offline', 'online', 'master', 'remote', 'legacy', 'local', 'alpha', 'beta', 'demo', 'main', 'test', 'gov', 'new', 'old', 'uat').

**Comment by mdmintz** [comment, marked as answer]
I'll add in ``--env=replica`` for the next release. All of them are one word. I'll keep it consistent.

If you need more customization, use `--data`, `--var1`, `--var2`, `--var3`, or `--variables=DICT`.

**Reply by AvinashK133** [reply]
Thank you for the quick response, can you please add  "stage, prod" as well, i know staging and production is already available, but we use and prefer the shorter words :) 
And is it possible to define our own set of environments(overriding the default ones), this will be very helpful.

**Reply by mdmintz** [reply]
`staging` and `production` are already available. We won't create duplicates that are shorter. That can lead to typos when setting and/or using an environment. Use the other variables if you want a custom environment to set: `--data`, `--var1`, `--var2`, `--var3`, or `--variables=DICT`.

**Reply by mdmintz** [reply]
"replica" was added as an env option in `4.21.0`, now available: https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.21.0

**Comment by AvinashK133** [comment]
Hi,
I tried the --env=replica with seleniumbase version 4.21.0 and latest  also but it is not working.
Please refer the attached screenshot below, meanwhile can share the workaround for this if possible.
Thanks in advance.

![image](https://github.com/seleniumbase/SeleniumBase/assets/112677932/78e307a5-b258-40c6-8209-3c668f838fb0)

**Reply by mdmintz** [reply]
It’s there on the latest version:

https://github.com/seleniumbase/SeleniumBase/blob/31020e4ae9644596460e9606312f94bc420aff51/seleniumbase/fixtures/constants.py#L41

You may have multiple environments on your system.

**Comment by AvinashK133** [comment]
I have created a new venv and installed version 4.21.0 and tried their, but getting same error, only one version i have installed.
screenshot in previous msg was from the newly created venv

**Reply by mdmintz** [reply]
That’s not the latest version. Always use the latest version.

This is how you determine your version: (At runtime)

```python
from seleniumbase import __version__

print(__version__)
```

**Comment by AvinashK133** [comment]
Installed latest version 4.22.5 and tried, unfortunately getting same error. Or am i missing something here?
![image](https://github.com/seleniumbase/SeleniumBase/assets/112677932/c3afa292-2868-4be6-9d5a-769a0dcf3443)

**Reply by mdmintz** [reply]
Your python and pytest could be using different environments. Put the version script inside something that is run with pytest.

**Reply by AvinashK133** [reply]
I am inside venv and referring to same python interpreator and pytest, "replica" option is not available in env list, attaching the screenshot for reference.
"Put the version script inside something that is run with pytest." --> i did not get this, can you pls elaborate this in detail

Thank You.
![image](https://github.com/seleniumbase/SeleniumBase/assets/112677932/5479371e-db31-4aae-b00b-13f383a61dae)

**Reply by AvinashK133** [reply]
just now found that 'replica' is not added in pytest_plugin as a pytest option. 
![image](https://github.com/seleniumbase/SeleniumBase/assets/112677932/8f55a2f0-bc06-45b2-9bd6-204d297725cc)

**Reply by mdmintz** [reply]
@AvinashK133 OK, I see the missing environment option there. I’m on vacation until next week, so I’ll add it into a new release after I get back.

**Reply by AvinashK133** [reply]
Thank You!

**Reply by mdmintz** [reply]
It's ready. Upgrade to version `4.22.6` (or newer when available).
